comp/core/gui/guiimpl/views/private/js/javascript.js

```diff
@@ -18,11 +18,62 @@ function sendMessage(endpoint, data, method, callback, callbackErr){
     url: window.location.href + endpoint,
     type: method,
     data: data,
-    success: callback,
-    error: callbackErr
+    success: function(data, status, xhr) {
+      // cleaning error layout before updating layour 
+      $("#error").hide()
+      $("#logged_out").hide();
+
+      // Set Agent state to "connected"
+      $("#agent_status").html("Connected <br> to Agent");
+      $("#agent_status").removeClass("disconnected")
+      $("#agent_status").addClass("connected")
+
+      try {
+        callback(data, status, xhr)
+      } catch (error) {
+        console.log(error)
+      }
+    },
+    error: function(requestObject, error, errorThrown) {
+      try {
+        callbackErr(requestObject, error, errorThrown)
+      } catch (error) {
+        console.log(error)
+      }
+
+      // Set Agent state to "disconnected"
+      $("#agent_status").html("Not connected<br> to Agent");
+      $("#agent_status").removeClass("connected")
+      $("#agent_status").addClass("disconnected")
+
+      // Display error layout 
+      setError(requestObject.status, requestObject.responseText)
```

Comment:
We are displaying the error message from the server, but we are not showing the instructions with the information, is that on purpose?